app/models/user.rb

```diff
@@ -23,6 +23,7 @@ class User < ActiveRecord::Base
 
   before_create :set_token
   validates :time_format, inclusion: { in: TIME_FORMATS }
+  validates :external_id, presence: :true, unless: :integration?
```

Comment:
users without external_id cannot log in ... so they are bogus records that should be invalid ... and if they somehow can log in I'd like to understand why / have them report an issue